tensorflow/cc/gradients/image_grad.cc

```diff
@@ -86,6 +86,27 @@ Status ScaleAndTranslateGradHelper(const Scope& scope, const Operation& op,
 }
 REGISTER_GRADIENT_OP("ScaleAndTranslate", ScaleAndTranslateGradHelper);
 
+Status CropAndResizeGradHelper(const Scope& scope, const Operation& op,
+                                   const std::vector<Output>& grad_inputs,
+                                   std::vector<Output>* grad_outputs) {
+  DataType input_type;
+  string method;
+  TF_RETURN_IF_ERROR(
+      GetNodeAttr(op.node()->attrs(), "method", &method));
+  TF_RETURN_IF_ERROR(
+      GetNodeAttr(op.node()->attrs(), "T", &input_type));
```

Comment:
Should you check for allowed types like in the Python version? https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/image_grad.py#L107

Reply:
I think it'll check the type during we build the code, so it has no need to check the types in the c plus plus version.